src/Microsoft.ML.Data/Transforms/Hashing.cs

```diff
@@ -834,6 +1112,96 @@ private void AddMetaKeyValues(int i, DataViewSchema.Annotations.Builder builder)
+                // uints and ints. Thus, ulongs, longs, doubles, floats, and booleans are not supported.
+                if (srcType == typeof(ushort) || srcType == typeof(short) || srcType == typeof(sbyte) || srcType == typeof(byte))
+                {
+                    var castOutput = ctx.AddIntermediateVariable(NumberDataViewType.UInt32, "CastOutput", true);
+                    var castNode = ctx.CreateNode("Cast", srcVariable, castOutput, ctx.GetNodeName(opType), "");
+                    castNode.AddAttribute("to", NumberDataViewType.UInt32.RawType);
+                    murmurNode = ctx.CreateNode(opType, castOutput, murmurOutput, ctx.GetNodeName(opType), "com.microsoft");
+                }
+                else if (srcType == typeof(uint) || srcType == typeof(int) || srcType == typeof(ReadOnlyMemory<char>))
+                {
+                    murmurNode = ctx.CreateNode(opType, srcVariable, murmurOutput, ctx.GetNodeName(opType), "com.microsoft");
+                }
+                else
+                {
+                    return false;
+                }
+
+                murmurNode.AddAttribute("positive", 1);
+                var seed = _parent._columns[iinfo].Seed;
+                if (_parent._columns[iinfo].UseOrderedHashing)
+                    seed = Hashing.MurmurRound(seed, 0);
+                murmurNode.AddAttribute("seed", seed);
+
+                // masking is done via bitshifts, until bitwise AND is supported by Onnxruntime
+                opType = "BitShift";
+                string bitShiftOutput = ctx.AddIntermediateVariable(_dstTypes[iinfo], "BitShiftOutput");
+                var shiftValue = ctx.AddInitializer((ulong)(32 - _parent._columns[iinfo].NumberOfBits), false);
+                var shiftNode = ctx.CreateNode(opType, new[] { murmurOutput, shiftValue }, new[] { bitShiftOutput }, ctx.GetNodeName(opType), "");
+                shiftNode.AddAttribute("direction", "LEFT");
+
+                opType = "BitShift";
+                var outputVariable = dstVariable;
+                if(_srcTypes[iinfo].GetItemType() is KeyDataViewType)
+                    outputVariable = ctx.AddIntermediateVariable(_dstTypes[iinfo], "BitShiftOutput2");
+                shiftNode = ctx.CreateNode(opType, new[] { bitShiftOutput, shiftValue }, new[] { outputVariable }, ctx.GetNodeName(opType), "");
+                shiftNode.AddAttribute("direction", "RIGHT");
+
+                if (_srcTypes[iinfo].GetItemType() is KeyDataViewType)
+                {
+                    opType = "Cast";
+                    string castOutput = ctx.AddIntermediateVariable(NumberDataViewType.Int64, "CastOutput2");
+                    var castNode = ctx.CreateNode(opType, outputVariable, castOutput, ctx.GetNodeName(opType), "");
+                    var t = NumberDataViewType.Int64.RawType;
+                    castNode.AddAttribute("to", t);
+
+                    opType = "Add";
```

Comment:
>Add [](start = 30, length = 3)

Does ONNX not know how to apply `Add` to unsigned integer types?

Reply:
Yeah, it's not currently supported. Planning to request support and remove additional operators once that is done.